activerecord/lib/active_record/associations/association_scope.rb

```diff
@@ -129,38 +113,57 @@ def next_chain_scope(scope, table, reflection, connection, assoc_klass, foreign_
         scope = scope.joins(join(foreign_table, constraint))
       end
 
-      def add_constraints(scope, owner, assoc_klass, refl, tracker)
-        chain = refl.chain
-        scope_chain = refl.scope_chain
-        connection = tracker.connection
+      class ReflectionProxy < SimpleDelegator # :nodoc:
+        attr_accessor :next
+        attr_reader :alias_name
+
+        def initialize(reflection, alias_name)
+          super(reflection)
+          @alias_name = alias_name
+        end
+
+        def all_includes; nil; end
+      end
 
-        tables = construct_tables(chain, assoc_klass, refl, tracker)
+      def get_chain(refl, association, tracker)
+        name = refl.name
+        runtime_reflection = ActiveRecord::Reflection::RuntimeReflection.new(refl, association)
+        prev = runtime_reflection
```

Comment:
Yup I got that one - I haven't pushed yet though. Gotta get focusing on work but will finish these later today. :+1: Thanks!